packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -63,9 +63,11 @@ const getNextColumnHeaderIndexes = (key: string, indexes: GridColumnHeaderIndexC
   }
 };
 
-export const useGridKeyboardNavigation = (apiRef: GridApiRef): void => {
+export const useGridKeyboardNavigation = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'pagination'>,
```

Comment:
Should we do it like this when there are only one or two props used?
```suggestion
  { pagination }: Pick<GridComponentProps, 'pagination'>,
```

Reply:
> I'm removing the prop decomposition in the parameters to follow #6
And because it make it obvious that we are accessing the props and not the state for cases where both make sense.

I like the clarity to have every prop usage being very explicit.

Reply:
Ok, I'm ok with this but then let's just follow the rule that regardless we always have it as `props`, because there are places where you kept the prop decomp.

Reply:
In which places ? I probably made a mistake

Reply:
+1 for avoiding destructuring when we can. I'm also annoyed by the selectors that makes it hard to know when refactoring what's used where.

In the core we do theme.palette.primary.main with the full path systematically to make it easier to work on the theme's structure.

Reply:
@flaviendelangle here for example https://github.com/mui-org/material-ui-x/blob/3ca9645d7df64a44c51df69e78ff578e5a677c69/packages/grid/_modules_/grid/hooks/features/useGridComponents.tsx#L67